test/jasmine/tests/transform_filter_test.js

```diff
@@ -256,6 +256,21 @@ describe('filter transforms calc:', function() {
             );
         });
 
+        it('should honored set axis type', function() {
+            var out = _transform([Lib.extendDeep({}, _base, {
+                x: [1, 2, 3, 0, -1, -2, -3],
+                transforms: [{
+                    operation: '>',
+                    value: -1,
+                    filtersrc: 'x'
+                }]
+            })], {
+                xaxis: { type: 'category' }
+            });
+
+            _assert(out, [-2, -3], [3, 1], [0.3, 0.4]);
```

Comment:
@bpostlethwaite this is a very important test case.

Here, a user inputs numerical data on a category axis - meaning that the data items are cast to strings and drawn in the ordered that they are input. That is,

```js
// x
[1, 2, 3, 0, -1, -2, -3] 

// is plotted with x coordinates
[0, 1, 2, 3, 4, 5, 6] 
```